Src/BootCamp.Chapter/Lesson4.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using System.Globalization;
+
+namespace BootCamp.Chapter
+{
+    class Lesson4
+    {
+        public static void Demo()
+        {
+            GetInfoAndPrint();
+            GetInfoAndPrint();
+        }
+
+        static void GetInfoAndPrint()
+        {
+            Console.Write("Name: ");
+            string name = Console.ReadLine();
+
+            var age = PromptForInt("Age: ");
+
+            var weight = PromptForFloat("Weight (in kg): ");
+
+            var height = PromptForFloat("Height (in m): ");
+
+            double bmi = CalculateBmi(weight, height);
+
+            Console.WriteLine($"{ name } is { age } years old, his weight is { weight } kg and his height is { height } m. { name }'s BMI is { bmi }.");
+        }
+
+        public static float CalculateBmi(float weight, float height)
+        {
+            bool isValidWeight = true;
+            bool isValidHeight = true;
+
+            string errorMessage = "{0} cannot be equal or less than zero, but was {1}.";
+
+            if (weight <= 0)
+            {
+                isValidWeight = false;
+            }
+
+            if (height <= 0)
+            {
+                isValidHeight = false;
+            }
+
+            if (!isValidWeight || !isValidHeight)
+            {
+                Console.WriteLine("Failed calculating BMI. Reason:");
+
+                if (!isValidWeight)
+                {
+                    Console.WriteLine(string.Format(errorMessage, "Weight", weight));
+                }
+
+                if (!isValidHeight)
+                {
+                    Console.WriteLine(string.Format(errorMessage, "Height", height));
+                }
+
+                return -1;
+            }
+
+            return weight / (height * height);
+        }
+
+        public static float PromptForFloat(string message)
+        {
+            var invC = CultureInfo.InvariantCulture;
+
+            Console.WriteLine(message);
+            string input = Console.ReadLine();
+            bool isNumber = float.TryParse(input, NumberStyles.Float, invC, out float number);
+            
+            if (!isNumber && !string.IsNullOrEmpty(input))
+            {
+                Console.Write($"\"{input}\" is not a valid number.");
+                return -1;
```

Comment:
-1 is a magic arbitary number. Please use a constant for that to explain its' meaning to me and other readers.